File: platform/macos/monarch_macos/DockSide.swift
//
//  DockSide.swift
//  monarch
//
//  Created by Fernando Trigoso on 8/12/21.
//  Copyright © 2021 Fernando Trigoso. All rights reserved.
//

import Foundation

enum DockSide {
    case right, left, undock
}

let defaultDockSide: DockSide = .right

func dockFromString(dockString: String) -> DockSide {
    switch dockString {
    case "left":
        return .left
    case "right":
        return .right
    case "undock":
        return .undock
    default:
        return .right
    }
}

File: platform/windows/src/dock_side.cpp
#include "dock_side.h"

DockSide dockFromString(std::string dockString)
{
    if (dockString == "left") {
        return DockSide::left;
    }
    else if (dockString == "right") {
        return DockSide::right;
    }
    else if (dockString == "undock") {
        return DockSide::undock;
    }
    else {
        return DockSide::right;
    }
}
